finality-provider/service/app.go

```diff
@@ -84,7 +85,21 @@ func NewFinalityProviderAppFromConfig(
 	fpMetrics := metrics.NewFpMetrics()
 	poller := NewChainPoller(logger, cfg.PollerConfig, consumerCon, fpMetrics)
 
-	return NewFinalityProviderApp(cfg, cc, consumerCon, em, poller, fpMetrics, db, logger)
+	pubRandStore, err := store.NewPubRandProofStore(db)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initiate public randomness store: %w", err)
+	}
+
+	rndCommiter := NewDefaultRandomnessCommitter(
+		NewRandomnessCommitterConfig(cfg.NumPubRand, int64(cfg.TimestampingDelayBlocks), cfg.ContextSigningHeight),
+		NewPubRandState(pubRandStore),
+		consumerCon,
+		em,
+		logger,
+		fpMetrics,
+	)
+
+	return NewFinalityProviderApp(cfg, cc, consumerCon, em, poller, rndCommiter, fpMetrics, db, logger)
```

Comment:
Since we are doing this refactoring, do we also need to have the finality operator and block querier here?